testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -308,6 +309,62 @@ class MetricFrameworkTests : public CppUnit::TestFixture
         CPPUNIT_ASSERT_EQUAL(0, static_cast<int>(pCounterGauge->queryValue()));
     }
 
+
+    void Test_histogram_metric()
+    {
+        std::vector<BucketDef> bucketDefs = {{"le 2", 2}, {"le 4", 4}, {"le 8", 8} };
+        std::shared_ptr<HistogramMetric> pHistogram = std::make_shared<HistogramMetric>("requests.dist", "description", SMeasureCount, bucketDefs);
+
+        std::vector<__uint64> values;
+
+        //
+        // Verify that the correct number of buckets is returned
+        values = pHistogram->queryHistogramValues();
+        CPPUNIT_ASSERT_EQUAL(bucketDefs.size()+1, values.size());
+
+        //
+        // Add a measurement and make sure it goes into the correct bucket
+        int sumMeasurements = 0;
+
+        pHistogram->recordMeasurement(4);
+        sumMeasurements += 4;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, {0,1,0,0});
+
+        //
+        // Add same measurement and make sure the bucket increments
+        pHistogram->recordMeasurement(4);
+        sumMeasurements += 4;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, {0,2,0,0});
+
+        //
+        // Add a measurement that is beyond the highest bucket and make sure it goes in the
+        // last "inf" bucket
+        pHistogram->recordMeasurement(20);
+        sumMeasurements += 20;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, {0,2,0,1});
+
+        //
+        // Just to make sure, increment another bucket
+        pHistogram->recordMeasurement(8);
+        sumMeasurements += 8;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, {0,2,1,1});
+    }
+
+
+    void checkHistogramBucketResult(std::shared_ptr<IMetric> pMetric, int expectedSum, const std::vector<int> &expectedValues)
+    {
+        std::shared_ptr<HistogramMetric> pHistogram;
```

Comment:
minor: These lines should be deleted, and the pMetric used - so this code will mirror the calls the sink will perform.

Reply:
Removed cast to histogram pointer type in favor of IMetric interface use only.